processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -469,7 +469,8 @@ public boolean isAssignableTo(Type other) {
             // the current target accessor can also be a getter method.
             // The following if block, checks if the target accessor should be overruled by an add method.
             if ( cmStrategy == CollectionMappingStrategyPrism.SETTER_PREFERRED
-                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED ) {
+                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED
+                || cmStrategy == CollectionMappingStrategyPrism.TARGET_IMMUTABLE ) {
```

Comment:
No it is clear. And you are right `Only target add- Available` is wrong, that cannnot happen (at least with the current implementation).